### PR TITLE
adds array/reverse

### PR DESCRIPTION
**Pull request by MathiasPaumgarten**
This PR is a proposal to an array/reverse method that returns a copy of the array unlike [Array.prototype.reverse()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reverse) which reverses the elements in place and doesn't return anything. 

I find this method quite useful and therefore wanted to ask if we are pro merge?

### CODE REVIEW

File: src/array/reverse.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+define(function () {
+
+    /**
+     * Returns a copy of the array in reversed order.
+     */
+    function reverse(array) {
+        var copy = array.slice();
```
Comment (MathiasPaumgarten) [inline src/array/reverse.js]: I could use `mout/lang/clone` but opted for the shorter version to skip unnecessary dependencies. Do we want to keep this DRY and add `clone()`?

**Comment by millermedeiros** [comment]
Sounds good to me! Feel free to merge. I don't have access to a computer where I can test it, but looks good.

File: src/array/reverse.js (hunk 1 above)
Comment (millermedeiros) [inline src/array/reverse.js]: Sounds better to keep as is. Fewer dependencies. Code is simple enough

**Comment by MathiasPaumgarten** [comment]
I did merge and push tags for `0.11.1` to git. I thought I waited for a full `0.12` for some other issues that would be quickly merged. However I can't push to npm as that is your personal account. Maybe, if you have time, you could `git pull origin master` and `npm publish` from your account @millermedeiros

**Pull request merged**